### ISSUE TITLE
If route id is empty and has auxiliary route next to it after reload error is thrown

### ISSUE THREAD

**Issue by BlindDespair** [angular/angular#24196]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior

When we have a route with child which specifies id and an auxiliary route and then we combine a url which goes has empty id in it and auxiliary url path next to it, then after reload it throws an error, even though if such url is combined while we use application it's working just fine.
Url like /user/(modal:hello)
Slash after user means that there is id expected but nothing was selected yet.

## Expected behavior

It should work after reload same as when we work with the routing while using application and just clicking buttons/links. 

## Minimal reproduction of the problem with instructions

https://stackblitz.com/edit/angular-playground-kdu4vs?file=app/app.component.html
1. Open the embedded app, click user link -> URL is changed to https://angular-playground-kdu4vs.stackblitz.io/user/ and component with text 'user works!' is displayed.
2. Click hello link -> URL is changed to https://angular-playground-kdu4vs.stackblitz.io/user/(modal:hello) and component with text 'user works!' is displayed and component with text 'Hello !' is displayed.
3. Reload embedded app -> ERROR in console. Result should be same as for step 2.

## What is the motivation / use case for changing the behavior?

Routing is used to items from list and display component of it's editor next to it. If nothing is selected then list is displayed but editor is not, empty id is used for that purpose. Why we don't use different routes is because it's basically same screen and redirecting is no go because it triggers components to reinitialize which leads to blinking UI issues. Auxiliary routes are used for feature modal windows.

## Environment

<pre><code>
Angular version: 6.0.0

Browser:
- [x] Chrome (desktop) version 66.0.3359.139
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 8.11.1  
- Platform: Windows 10  

Others:

</code></pre>

**Comment by BlindDespair** [comment]
For our uses case I solved it by using custom RouteReuseStrategy posted in one of the issues regarding component reuse in routing. 
But I think this issue still makes sense as it should not throw error after reload if doesn't throw it on user interactions.

**Comment by jasonaden** [comment]
I believe this is expected, since path params would be for required parameters. If it's an optional param, it should be a matrix param. Does that clarify?

**Comment by BlindDespair** [comment]
@jasonaden I would disagree here because we have same url "http://localhost:232/app/users/(modal:info)"
which works while app is running and doesn't when we reload it. Imo it should behave same in both cases either work or not work, but should be consistent.

**Comment by jasonaden** [comment]
I get what you're saying here. I believe the correct behavior here would be to error when initially navigating, so you don't end up on a URL that can't be parsed. Or to not allow serialization of `null/undefined` path params.

URL serialization needs to happen without the context of the configured routes. It needs to be able to serialize any set of `UrlSegment`s with no additional knowledge. Also, when clicking a `routerLink`, we call `navigateByUrl` with an already parsed `UrlTree`. The thing is a `UrlTree` can be built with `null` path params, but the parsing logic cannot handle that serialized URL.

Any additional thoughts?

**Comment by jasonaden** [comment]
Closing as working as intended and no response on above. If this still appears to be an issue that should be addressed, please create a new issue and mention the relation to this issue.

**Comment by BlindDespair** [comment]
@jasonaden Sorry for late response, but I agree with what you said 7 days ago. It's fine that issue is closed now for me, since I am not blocked by it, we managed to solve problem in a better way. Thanks. But if in future it would be consistent and navigating to url with null/undefined params would lead to error, it would be great.
